lib/install/management/appliance.go

```diff
@@ -490,21 +485,31 @@ func (d *Dispatcher) createAppliance(conf *config.VirtualContainerHostConfigSpec
 		return err
 	}
 
-	var info *types.TaskInfo
-	// create appliance VM
-	if d.isVC && d.vchVapp != nil {
-		info, err = tasks.WaitForResult(d.op, func(ctx context.Context) (tasks.Task, error) {
-			return d.vchVapp.CreateChildVM(ctx, *spec, d.session.Host)
-		})
-	} else {
-		// if vapp is not created, fall back to create VM under default resource pool
-		info, err = tasks.WaitForResult(d.op, func(ctx context.Context) (tasks.Task, error) {
-			return d.session.VMFolder.CreateVM(ctx, *spec, d.vchPool, d.session.Host)
-		})
+	// Create the VCH inventory folder
+	if d.isVC {
+		folderPath := fmt.Sprintf("%s", path.Join(d.session.VMFolder.InventoryPath, conf.Name))
+
+		d.op.Info("Creating the VCH folder")
+		// update the session pointer with the VCH Folder
+		d.session.VCHFolder, err = d.session.VMFolder.CreateFolder(d.op, spec.Name)
+		if err != nil {
+			if soap.IsSoapFault(err) {
+				switch soap.ToSoapFault(err).VimFault().(type) {
+				case types.DuplicateName:
```

Comment:
Does the "an existing object with the same name" has any meaning for the end user? What kind of the object? Can we make it a little friendlier?

Reply:
e.g. "The specified VCH name is already in use."

Reply:
Nicely done...I like that...thanks!